### PR TITLE
create progress bar component

### PR DESCRIPTION
**Pull request by akinwale**
Replaces use of the platform-dependent `ProgressBarAndroid` component.

### CODE REVIEW

File: app/src/component/progressBar/view.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+class ProgressBar extends React.PureComponent {
+  static propTypes = {
+    borderRadius: PropTypes.number,
+    color: PropTypes.string.isRequired,
+    height: PropTypes.number,
+    progress: function(props, propName, componentName) {
+      const value = parseInt(props[propName], 10);
+      if (isNaN(value) || props[propName] < minProgress || props[propName] > maxProgress) {
+        return new Error('progress should be between 0 and 100');
```
Comment (kauffj) [inline app/src/component/progressBar/view.js]: Should this return an error or throw one?

File: app/src/styles/fileList.js [hunk 2]
```diff
@@ -71,9 +71,6 @@ const fileListStyle = StyleSheet.create({
   },
   progress: {
     marginTop: (screenWidthPixels <= 720) ? 2 : 4,
-    height: 3,
```
Comment (kauffj) [inline app/src/styles/fileList.js]: Outside of this PR, but if 720 is a special value that appears in multiple places it should be a constant.

File: app/src/component/progressBar/view.js (hunk 1 above)
Comment (akinwale) [reply app/src/component/progressBar/view.js]: According to https://reactjs.org/docs/typechecking-with-proptypes.html, return.

**Comment by akinwale** [comment]
> Basically looks good. Out of curiosity, what was wrong with `ProgressBarAndroid` from ReactNative?

It is Android specific, so it wouldn't work on iOS. I'd have to use the `ProgressViewIOS` too, which would require adding additional platform checks in the code. Made sense to just use a single component without having to perform these checks.

**Pull request merged**